examples/components/http-hello-world/README.md

```diff
@@ -0,0 +1,184 @@
+# WASI `http-hello-world` in JavaScript
+
+This folder contains a WebAssembly Javascript component that uses [`wasi:http`][wasi-http] for enabling HTTP handlers in Javascript.
+
+It uses [`jco`][jco] to:
+
+- Generate a WebAssembly component (via `jco componentize`) that can be executed by a WebAssembly runtime (ex. [`wasmtime serve`][wasmtime])
+
+[nodejs]: https://nodejs.org
+[jco]: https://bytecodealliance.github.io/jco/
+[wasi-http]: https://github.com/WebAssembly/wasi-http
+[wasmtime]: https://github.com/bytecodealliance/wasmtime
+
+# Quickstart
+
+## Dependencies
+
+First, install required dependencies:
+
+```console
+npm install
+```
+
+> [!NOTE]
+> As this is a regular NodeJS project, you can use your package manager of choice (e.g. `yarn`, `pnpm`)
+
+At this point, since this project is *just* NodeJS, you could use the module from any NodeJS project or browser project where appropriate.
+
+That said, we'll be focusing on building the JS code we've written so far into a WebAssembly binary, which can run *anywhere* WebAssembly runtimes are supported,
+including in other languages, and the browser.
+
+## Building the WebAssembly component
+
+We can build a WebAssembly component binary out of this JS project with `jco`:
+
+```console
+npm run build
+```
+
+A WebAssembly binary will be written to `string-reverse.wasm`.
+
+## Serving web requests with the WebAssembly component
+
+To run the component and serve requests we can either use `jco` or `wasmtime`:
+
+```console
+$ jco serve http-hello-world.wasm
+Server listening on 8000...
+```
+
+Similarly you can also use `wasmtime`:
+
+```
+$ wasmtime serve -S common http-hello-world.wasm
+Serving HTTP on http://0.0.0.0:8080/
+```
+
+With either approach, you can use `curl` the appropriate URL to trigger your WebAssembly component.
+
+> [!NOTE]
+> The implementations of `jco serve` and `wasmtime serve` are what actually *fulfill* all the imports
+> of your component (see combined/merged `world root` above), and use the `wasi:http/incoming-handler` *export*
+> to make web serving actually happen.
+
+# How it works
+
+## Exploring this Component WIT
+
+As WebAssembly components are powered by a [WebAssembly Interface Types ("WIT")][wit]-first workflow, making
+a HTTP handler component in WebAssembly requires creating a WIT contract to represent that component.
+
+This folder contains a `wit` directory (by convention) with the following content in `component.wit`:
+
+```wit
+package example:http-hello-world;
+
+world component {
+    export wasi:http/incoming-handler@0.2.0;
```

Comment:
Should we be using WASI@0.2.1? Or 0.2.2?

Thinking about this - we likely will need a process to update the examples to newer WASI versions.

Reply:
Ah thanks! will update this. In general we should try to use the latest, but also I don't expect things to just break if we're one or two minor versions behind.

Will update to 0.2.2 here!